packaging/package-indexer/indexer/rpm_indexer.py

```diff
@@ -0,0 +1,265 @@
+        dist_dir: Path,
+        cdn: CDN,
+        gpg_key_path: Path,
+        gpg_key_passphrase: Optional[str],
+    ) -> None:
+        self.__gpg_key_path = gpg_key_path
+        self.__gpg_key_passphrase = gpg_key_passphrase
+        super().__init__(
+            incoming_remote_storage_synchronizer=incoming_remote_storage_synchronizer,
+            indexed_remote_storage_synchronizer=indexed_remote_storage_synchronizer,
+            incoming_sub_dir=incoming_sub_dir,
+            indexed_sub_dir=Path("rpm", dist_dir),
+            cdn=cdn,
+        )
+
+    def __move_files_from_incoming_to_indexed(self, incoming_dir: Path, indexed_dir: Path) -> None:
+        for file in filter(lambda path: path.is_file(), incoming_dir.rglob("*")):
+            relative_path = file.relative_to(incoming_dir)
+            platform = relative_path.parent
+            file_name = relative_path.name
+
+            new_path = Path(indexed_dir, platform, "x86_64", file_name)
+
+            self._log_info("Moving file.", src_path=str(file), dst_path=str(new_path))
+
+            new_path.parent.mkdir(parents=True, exist_ok=True)
+            utils.move_file_without_overwrite(file, new_path)
+
+    def _prepare_indexed_dir(self, incoming_dir: Path, indexed_dir: Path) -> None:
+        self.__move_files_from_incoming_to_indexed(incoming_dir, indexed_dir)
+
+    def __get_gpg_key_email(self, gnupghome_dir: Path) -> str:
+        with TemporaryFile(mode="w+", encoding="utf-8") as f:
+            utils.execute_command(
+                ["gpg", "--list-secret-keys", "--with-colons"],
+                stdout=f,
+                env={"GNUPGHOME": str(gnupghome_dir)},
+            )
+            f.seek(0)
+
+            for line in f.readlines():
+                if line.startswith("uid"):
+                    return line.split(":")[9]
+
+        raise Exception("Failed to get GPG key email.")
+
+    def __rpm_addsign(self, gnupghome_dir: Path, indexed_dir: Path) -> None:
+        email = self.__get_gpg_key_email(gnupghome_dir)
+
+        sign_cmd = r"%{__gpg} gpg --no-tty --no-options --batch "
+        if self.__gpg_key_passphrase is not None:
+            sign_cmd += "--pinentry-mode loopback --passphrase-fd 0 "
+        sign_cmd += r"--no-armor -u \"%{_gpg_name}\" -sbo %{__signature_filename} %{__plaintext_filename}'"
+
+        rpmmacros_path = Path(gnupghome_dir, ".rpmmacros").absolute()
+        rpmmacros_path.write_text(
+            "%_signature gpg\n"
+            f"%_gpg_name {email}\n"
+            f"%_gpg_path {str(gnupghome_dir)}\n"
+            f"%__gpg {shutil.which('gpg')}\n"
+            f"%__gpg_sign_cmd {sign_cmd}\n"
+        )
+
+        for rpm_file in indexed_dir.rglob("*.rpm"):
+            self._log_info("Signing rpm.", rpm_file=str(rpm_file))
+            utils.execute_command(
+                ["rpm", "--macros", str(rpmmacros_path), "--addsign", str(rpm_file.absolute())],
+                input=self.__gpg_key_passphrase,
+            )
```

Comment:
gpg key with passphrase does not work.

probably we need to use a gpg-agent session, so passphrase input is not needed for each sign.